core/src/retransmit_stage.rs

```diff
@@ -319,28 +323,35 @@ fn retransmit_shred(
         cluster_nodes.get_retransmit_addrs(slot_leader, key, root_bank, data_plane_fanout)?;
     let addrs: Vec<_> = addrs
         .into_iter()
-        .filter(|addr| ContactInfo::is_valid_address(addr, socket_addr_space))
+        .filter(|addr| socket_addr_space.check(addr))
```

Comment:
Maybe I'm missing something, but this change doesn't seem related. Is it ok to drop the additional checks from `ContactInfo::is_valid_address`? Or did you plan to add more checks to `SocketAddrSpace::check`?

Reply:
This was left out of https://github.com/solana-labs/solana/pull/31248
The new contact-info api already does sanitize the socket address before returning, so `is_valid_address` check here is redundant.

Reply:
Great, thanks for the background